syntax = "proto3";
package key;





/*
ListCollectorKey请求
*/
message ListCollectorKeyRequest {
    /*
    页码
    */
    int32 page = 1 ;
    /*
    每页大小
    */
    int32 pageSize = 2 ;
}

/*
ListCollectorKey返回
*/
message ListCollectorKeyResponse {
message List {
message TagDefine {
    /*
    名字
    */
    string name = 1 ;
    /*
    默认值
    */
    string default = 2 ;
    /*
    是否只读
    */
    bool readOnly = 3 ;
}
        
message ParamDefine {
    /*
    名字
    */
    string name = 1 ;
    /*
    值类型
    */
    string valueType = 2 ;
    /*
    默认值
    */
    string default = 3 ;
    /*
    是否只读
    */
    bool readOnly = 4 ;
}
        
    
    /*
    唯一名称
    */
    string name = 1 ;
    /*
    key名
    */
    string keyName = 2 ;
    /*
    标签
    */
    repeated string labels = 3 ;
    /*
    指标id
    */
    string keyId = 4 ;
    /*
    版本
    */
    int32 version = 5 ;
    /*
    key值
    */
    string key = 6 ;
    /*
    类型
    */
    string metricType = 7 ;
    /*
    数据类型
    */
    string dataType = 8 ;
    /*
    所属Agent
    */
    string agentType = 9 ;
    /*
    维度定义
    */
    repeated TagDefine tagDefine = 10 ;
    /*
    参数定义
    */
    repeated ParamDefine paramDefine = 11 ;
    /*
    帮助
    */
    string help = 12 ;
    /*
    单位
    */
    string unit = 13 ;
    /*
    采集指标实例版本号
    */
    int32 metricVersion = 14 ;
}
        
    
    /*
    页数
    */
    int32 page = 1 ;
    /*
    页大小
    */
    int32 page_size = 2 ;
    /*
    总数
    */
    int32 total = 3 ;
    /*
    数据列表
    */
    repeated List list = 4 ;
}

/*
ListCollectorKeyApi返回
*/
message ListCollectorKeyResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    key.ListCollectorKeyResponse data = 4 ;
}

